examples/http/azure_containers.js

```diff
@@ -0,0 +1,552 @@
+        if (key !== "timeStamp") {
+            return key;
+        }
+    }
+    return null;
+}
+
+/**
+ * Extracts performance metrics for a Container Group and populates the Container Group info object.
+ * @param {Object} performanceInfo - The performance metrics information for the Container Group.
+ * @param {Object} containerGroupInfo - The Container Group information object to populate with performance metrics.
+ */
+function extractContainerGroupPerformance(performanceInfo, containerGroupInfo) {
+    if (performanceInfo.name.value) {
+        if (performanceInfo.timeseries && performanceInfo.timeseries[0] && performanceInfo.timeseries[0].data) {
+            const key = getNonTimeStampKey(performanceInfo.timeseries[0].data);
+            containerGroupInfo[performanceInfo.name.value] = key ? performanceInfo.timeseries[0].data[0][key] : "N/A";
+        } else {
+            containerGroupInfo[performanceInfo.name.value] = "N/A"
+        }
+    }
+}
+
+/**
+ * Processes the response from the Container Groups API call and populates the table with Container Group data.
+ * @param {Object} d - The deferred promise object.
+ * @param {Object} containerGroupInfo - The Container Group information object containing resource group and Container Group details.
+ * @returns {Function} A function to process the HTTP response.
+ */
+function processContainerGroupPerformanceResponse(d, containerGroupInfo) {
+    return function process(error, response, body) {
+        checkHTTPError(error, response);
+        if (error || response.statusCode !== 200) {
+            return;
+        }
+        try {
+            const bodyAsJSON = JSON.parse(body);
+            if (!bodyAsJSON.value) {
+                D.failure(D.errorType.GENERIC_ERROR)
+                return;
+            }
+            bodyAsJSON.value.map(function (performanceInfo) {
+                extractContainerGroupPerformance(performanceInfo, containerGroupInfo)
+            });
+            d.resolve(containerGroupInfo);
+        } catch (parseError) {
+            console.error("Error parsing Container Groups configuration:", parseError);
+            d.reject("Failed to parse response for " + containerGroupInfo.name);
+        }
+    }
+}
+
+/**
+ * Retrieves performance metrics for a list of Container Groups.
+ * @param {Array} containerGroupInfoList - The list of Container Group information objects.
+ * @returns {Promise} - A promise that resolves with an array of vContainer Groupirtual machine information objects.
+ */
+function retrieveContainerGroupsPerformanceMetrics(containerGroupInfoList) {
+    const performanceKeyGroups = [];
+    const maxGroupSize = 20;
+    for (let i = 0; i < performanceMetrics.length; i += maxGroupSize) {
+        performanceKeyGroups.push(performanceMetrics.slice(i, i + maxGroupSize).map(function (metric) {
+            return metric.key
+        }).join(','));
+    }
+    const promises = containerGroupInfoList.map(function (containerGroupInfo) {
```

Comment:
please for promises use D.q and do not use Promise

create a promies variable as empty array and iterate over containerGroupInfoList and then iterate over performanceKeyGroups and in the second loop create a defer and push d.promise to your promises table